contrib/src/main/java/com/datatorrent/contrib/parquet/AbstractParquetFileReader.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.parquet;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.hadoop.fs.Path;
+
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.lib.io.fs.AbstractFileInputOperator;
+
+import parquet.example.data.Group;
+import parquet.hadoop.ParquetReader;
+import parquet.hadoop.example.GroupReadSupport;
+import parquet.schema.MessageType;
+import parquet.schema.MessageTypeParser;
+
+/**
+ * Base implementation of ParquetFileReader. Reads parquet files from input
+ * directory using GroupReadSupport. Derived classes need to implement
+ * {@link #convertGroup(Group)} method to convert Group to other type. Example
+ * of such implementation is {@link ParquetFilePOJOReader} that converts Group
+ * to POJO.
+ * 
+ * @since 3.3.3
+ */
+public abstract class AbstractParquetFileReader<T> extends AbstractFileInputOperator<T>
```

Comment:
Are there places other than ParquetFilePOJOReader.java where this this abstract class can be extended... If not can we merge these 2 files. Just want to avoid management of multiple files.